frontend/src/scenes/Apps/containers/AppEditor/index.tsx

```diff
@@ -23,17 +23,24 @@ import { IResource } from '../../../../common/models/resources/IResource';
 import { Navbar, NavDropdown, Nav, Form, Button, Image, Col } from 'react-bootstrap';
 import { Routes } from '../../../../common/enums/Routes';
 import { editAppRoutine, fetchSelectAppRoutine, setNewAppNameRoutine } from '../../routines';
+import { isAccess } from '../../../../common/helpers/permissionHelper';
+import { Roles } from '../../../../common/enums/UserRoles';
 
 interface IProps {
   resources: Array<IResource>,
-  fetchResources: () => void
+  fetchResources: () => void,
 }
 
 const AppEditor: React.FC<IProps> = ({ resources, fetchResources }) => {
   const { id }: IFetchParams = useParams();
+  const history = useHistory();
   useEffect(() => {
+    if (!isAccess([Roles.Admin, Roles.Developer])) {
+      history.push('/apps');
```

Comment:
it would be better to use history.replace() and pass corresponding app preview route as parameter